assets/js/googlesitekit/datastore/user/user-info.js

```diff
@@ -0,0 +1,256 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+
+const { createRegistrySelector } = Data;
+
+const RECEIVE_USER_INFO = 'RECEIVE_USER_INFO';
+const RECEIVE_USER_IS_VERIFIED = 'RECEIVE_USER_IS_VERIFIED';
+
+const INITIAL_STATE = {
+	user: undefined,
+	verified: undefined,
+};
+
+export const actions = {
+	/**
+	 * Stores user info in the datastore.
+	 *
+	 * Because this is frequently-accessed data, this is usually sourced
+	 * from a global variable (`_googlesitekitUserData`), set by PHP
+	 * in the `before_print` callback for `googlesitekit-datastore-user`.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} userInfo User info, usually supplied via a global variable from PHP.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveUserInfo( userInfo ) {
+		invariant( userInfo, 'userInfo is required.' );
+		const { user, verified } = userInfo;
+		return {
+			payload: {
+				user,
+				verified,
+			},
+			type: RECEIVE_USER_INFO,
+		};
+	},
+
+	/**
+	 * Stores user verification status in the datastore.
+	 *
+	 * Because this is frequently-accessed data, this is usually sourced
+	 * from a global variable (`_googlesitekitUserData`), set by PHP
+	 * in the `before_print` callback for `googlesitekit-datastore-user`.
+	 *
+	 * @since n.e.x.t
+	 * @private
+	 *
+	 * @param {Object} userIsVerified User verification status, usually supplied via a global variable from PHP.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveUserIsVerified( userIsVerified ) {
+		invariant( userIsVerified, 'userIsVerified is required.' );
+		const { verified } = userIsVerified;
+		return {
+			payload: {
+				verified,
+			},
+			type: RECEIVE_USER_IS_VERIFIED,
+		};
+	},
+};
+
+export const controls = {};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case RECEIVE_USER_INFO: {
+			const { user } = payload;
+			return {
+				...state,
+				user,
+			};
+		}
+		case RECEIVE_USER_IS_VERIFIED: {
+			const { verified } = payload;
+			return {
+				...state,
+				verified,
+			};
+		}
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getUser() {
+		const { select } = yield Data.commonActions.getRegistry();
+
+		if ( select( STORE_NAME ).getUser() !== undefined ) {
+			return;
+		}
+
+		if ( ! global._googlesitekitUserData ) {
+			global.console.error( 'Could not load core/user info.' );
+			return;
+		}
+
+		yield actions.receiveUserInfo( { ...global._googlesitekitUserData } );
```

Comment:
Regarding my previous comment, this should maybe become `actions.receiveUser( global._googlesitekitUserData.user )`.